### PR TITLE
Add Cognito authorization vi Amplify sdk and related compensating changes.

### PR DESCRIPTION
**Pull request by eriksendc**
Summary
---------

Created an AWS Cognito user pool, and there is now authentication to the web app and the api calls the web app makes.

Test Plan
---------

1. terraform destroy
2. terraform apply
3. npm start
4. exercise /matches and /upload

### CODE REVIEW

File: hasher-matcher-actioner/terraform/api/main.tf
Comment (eriksendc) [inline hasher-matcher-actioner/terraform/api/main.tf:109]: Note that I didn't delete the python source for the authorizer... sort of on purpose. was thinking to ask for comments on that.

File: hasher-matcher-actioner/terraform/api/main.tf
Comment (BarrettOlson) [inline hasher-matcher-actioner/terraform/api/main.tf:168]: did you mean to pass the variables all the way down to here?

File: hasher-matcher-actioner/terraform/api/main.tf
Comment (BarrettOlson) [inline hasher-matcher-actioner/terraform/api/main.tf:108]: nice! can you delete the python code too?

File: hasher-matcher-actioner/terraform/api/main.tf
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/api/main.tf:168]: Doh! Yes I set up all those nice variables and then didn't use them!!!!

**Review by BarrettOlson — APPROVED** [review]
Accepting to unblock. 

Please either update contributing and/or webapp/README with the steps needed to be able to run your test plan or add a Make command to do it ~automatically before merge (the .ENV stuff). 

And like @Dcallies said the sooner we can make even these cognito pieces ~easily deployable the better. :) Thanks again for working on this!

File: hasher-matcher-actioner/terraform/api/main.tf
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/api/main.tf]: Fixed!

File: hasher-matcher-actioner/terraform/api/main.tf
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/api/main.tf:109]: We decided to delete api_auth.py.

File: hasher-matcher-actioner/terraform/api/main.tf
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/api/main.tf:108]: absolutely!

File: hasher-matcher-actioner/webapp/README.md
Comment (BarrettOlson) [inline hasher-matcher-actioner/webapp/README.md:9]: ❤️, think is it worth updating CONTRIBUTING.md with something like "For the UI/Webapp components see `webapp/README` ?

File: hasher-matcher-actioner/terraform/authentication/main.tf
Comment (BarrettOlson) [inline hasher-matcher-actioner/terraform/authentication/main.tf:50]: Just curious, how do these work? can we support them both locally and using cloudfront? Thanks!

**Pull request merged**
